### ISSUE TITLE
UA and WA is inverted

### ISSUE THREAD

**Issue by bagustris** [sinhat98/adapter-wavlm#6]
@sinhat98,

I read the paper and found that UA and WA are inverted (both definition and coding in this report).

From the paper (page 3 left bottom):

> The weighted accuracy (WA) is the average of the classification accuracy per class

WA should be the overall accuracy, not the average accuracy. The average accuracy (or balanced accuracy) should be named UA (unweighted accuracy) or UAR (unweighted average recall).

Reference (there are some, but I cited one below):
Xu, H., Zhang, H., Han, K., Wang, Y., Peng, Y., & Li, X. (2019). Learning Alignment for Multimodal Emotion Recognition from Speech. Proc. Interspeech 2019, 3569–3573. https://doi.org/10.21437/Interspeech.2019-3247

On page 3, the authors wrote:

> weighted accuracy (WA) that is the overall classification accuracy and unweighted accuracy (UA) that is the average recall over the emotion categories.

So, in both the paper and this repo, UA showed WA and WA showed UA. Do you agree with this?

**Comment by sinhat98** [comment]
@bagustris 
I agree that the common understanding between WA and UA aligns closely with our discussion. I apologize for any confusion. 
I'd like to refer to a series of experiments detailed in a paper on data preparation, available at https://devblogs.microsoft.com/ise/2015/11/29/data-preparation-the-balancing-act/, which defines WA and UA as we've discussed. 
It's important to note that while this paper maintains a consistent definition of WA and UA, I acknowledge that in many other papers, these terms might carry the interpretation you've highlighted.

**Comment by bagustris** [comment]
@sinhat98 
Thanks for making it clear.
Regarding the Microsoft article, I have already made an issue in their GitHub repo since last year (the link is on the bottom page of the article you shared).

https://github.com/ryubidragonfire/Emotion/issues/3
